src/components/input/Select.tsx

```diff
@@ -604,13 +607,17 @@ function SelectMain<T>({
           listboxOverflow,
         }}
       >
-        <ul
+        <div
           className={classnames(
             'absolute z-5 max-h-80 overflow-y-auto overflow-x-hidden',
-            // We don't want the listbox to ever render outside the viewport,
-            // and we give it a 16px gap
-            'max-w-[calc(100%-16px)]',
             'rounded border bg-white shadow hover:shadow-md focus-within:shadow-md',
+            listboxAsPopover && [
+              // We don't want the listbox to ever render outside the viewport,
+              // and we give it a 16px gap
+              'max-w-[calc(100%-16px)]',
+              // Overwrite [popover] default styles
```

Comment:
What are the `popover` default styles?

Reply:
> Surprisingly, it didn't have any effect when `[popover]` was used in a `ul` element. I suspect it was because of the style resets that tailwind sets.

Yep, exactly.

![image](https://github.com/user-attachments/assets/6007651c-a4bf-4ca2-8275-e6d2ef410459)